### ISSUE TITLE
upgrade netty.tcnative to 2.0.0.Final to overcome collisions

### ISSUE THREAD

**Issue by frko** [grpc/grpc-java#2936]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
1.2.0 ( in google-cloud-java )

### What JVM are you using (`java -version`)?
8

### What did you do?
If possible, provide a recipe for reproducing the error.
included gcloud-java with rpc into an context that already uses tcnative - spring boot in this case

### What did you expect to see?
the application should be able to find and use the native driver

### What did you see instead?
An incarnation of the following issue -
https://github.com/netty/netty-tcnative/issues/151

**Comment by carl-mastrangelo** [comment]
tcnative is dependent on 4.1.9+ right?  I believe there are showstopping bugs in the latest netty which prevent upgrading.  You request has been noted; we like staying up to date with the latest and greatest.

**Comment by frko** [comment]
Seems to be the case indeed.
https://github.com/netty/netty-tcnative/blob/netty-tcnative-parent-2.0.0.Final/pom.xml
```
<parent>
    <groupId>io.netty</groupId>
    <artifactId>netty-parent</artifactId>
    <version>4.0.18.Final</version>
  </parent>
```

**Comment by frko** [comment]
Also  created the following google cloud java issue as its the way I encountered the issue.
https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1958

**Comment by ejona86** [comment]
This was fixed with #2973
